collect_app/src/test/java/org/odk/collect/android/http/mock/MockHttpClientConnection.java

```diff
@@ -0,0 +1,68 @@
+package org.odk.collect.android.http.mock;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.odk.collect.android.http.HttpCredentialsInterface;
+import org.odk.collect.android.http.HttpGetResult;
+import org.odk.collect.android.http.HttpHeadResult;
+import org.odk.collect.android.http.OpenRosaHttpInterface;
+import org.odk.collect.android.utilities.ResponseMessageParser;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class MockHttpClientConnection implements OpenRosaHttpInterface {
```

Comment:
Same for this! This can be added to the `TestModule` as well and when the `CollectServerClient` is being created it would be supplied as a dependency.

Reply:
Done 👍